src/java/org/apache/cassandra/cql3/QueryProcessor.java

```diff
@@ -149,11 +153,13 @@ public void preloadPreparedStatements()
                     clientState.setKeyspace(keyspace);
 
                 Prepared prepared = parseAndPrepare(query, clientState, false);
```

Comment:
`parseAndPrepare` might return Prepared with initialized size already so everything which will be needed after this like would be `preparedStatements.put(id, prepared)`

Reply:
Yes. And with that the change is much less invasive.